lib/switchtec.c

```diff
@@ -1553,8 +1577,24 @@ int switchtec_parse_log(FILE *bin_log_file, FILE *log_def_file,
 	/* parse each log entry */
 	while (fread(&log_data, sizeof(struct log_a_data), 1,
 		     bin_log_file) == 1) {
+		fw_version = fw_version_log? fw_version_log:fw_version_def;
```

Comment:
Generally prefer spaces around the ? and : in the ternary operator.

But I also try to avoid the ternary operator except in rare cases.

I might just do:

```
if (fw_version_log)
    gen_file = switchtec_fw_version_to_gen(fw_version_log);
else
    gen_file = switchtec_fw_version_to_gen(fw_version_def);
```